frontend/components/MainPage/header/AgentButton/AgentButton.tsx

```diff
@@ -0,0 +1,61 @@
+import { Button } from 'antd';
+import { useMemo } from 'react';
+
+import { MiddlewareDeploymentStatus } from '@/client';
+import { useService } from '@/hooks/useService';
+import { useServices } from '@/hooks/useServices';
+import { useStakingContractInfo } from '@/hooks/useStakingContractInfo';
+
+import {
+  CannotStartAgentDueToUnexpectedError,
+  CannotStartAgentPopover,
+} from '../CannotStartAgentPopover';
+import { AgentNotRunningButton } from './AgentNotRunningButton';
+import { AgentRunningButton } from './AgentRunningButton';
+import { AgentStartingButton } from './AgentStartingButton';
+import { AgentStoppingButton } from './AgentStoppingButton';
+
+export const AgentButton = () => {
+  const { selectedService } = useServices();
+
+  const {
+    service,
+    deploymentStatus: serviceStatus,
+    isLoaded,
+  } = useService({ serviceConfigId: selectedService?.service_config_id });
+
+  const { isEligibleForStaking, isAgentEvicted } = useStakingContractInfo();
```

Comment:
no particular pattern enforced

generally i add spaces when there are many hooks, or input params that the formatter has line broken. helps me maintain focus on the item i'm working on..

perhaps we can add a whitespace eslint rule so this can be mass fixed/enforced?